Making group

oYasuiHaruki opened the pull request:
# what
グループ機能の搭載

# why
chat-spaceにグループ機能が必要なため

oYasuiHaruki committed:
groupのコントローラー、モデル、ビューの作成
migrationファイルを編集して、アソシエーションを作成

oYasuiHaruki committed:
グループ作成機能の実装

oYasuiHaruki committed:
migrationファイルにオプションを追加

oYasuiHaruki committed:
モデルファイルの編集

oYasuiHaruki committed:
index.html.erbの編集

yasui83 commented:
# what
グループ作成機能の実装を行いました
まだ、編集機能はつけていません

1.groupテーブルと中間テーブルの作成
2.モデルファイルとマイグレーションファイルの作成
3.groupコントローラーの作成
4.トップ画面からグループ新規作成画面へ飛ぶように作成

# why
新規作成機能のみ搭載しました
まだ、編集機能は搭載されていないです

sato11 commented on app/assets/stylesheets/_clearfix.scss:
```diff
@@ -0,0 +1,5 @@
+    clear: both;
+    content: '';
+    display: block;
+}
```
ファイル末尾の改行忘れないようにしよう！

sato11 commented on app/assets/stylesheets/_clearfix.scss:
```diff
@@ -0,0 +1,5 @@
+.clearfix:after {
```
これは mixin フォルダを作ってその中で管理した方が良さそうだね。

sato11 commented on app/assets/stylesheets/_group.scss:
```diff
@@ -0,0 +1,106 @@
+
+    li {
+      margin: 0;
+      padding: 0;
```
この辺り margin: 0; とか padding: 0; とかわざわざ書いてるのってどんな意図がある？

sato11 commented on app/assets/stylesheets/_group.scss:
```diff
@@ -0,0 +1,106 @@
+  .chat-group-form__field {
+    margin-top: 30px;
+
+    .chat-group-form__field--left {
```
せっかく bem を使ってるんだから、全体的に&記法を活用してスマートに書いてみよう！

sato11 commented on app/controllers/group_controller.rb:
```diff
@@ -0,0 +1,21 @@
+
+  def edit
+
+  end
```
実装してないならアクションも書かない方がいいね。自分の意図とコードは一致させよう！

sato11 commented on app/controllers/group_controller.rb:
```diff
@@ -0,0 +1,21 @@
+
+    def group_params
+      params.require(:group).permit(:name, { :user_ids => [] })
+    end
```
ここインデント深くしなくて大丈夫だよ 👍

sato11 commented on app/controllers/group_controller.rb:
```diff
@@ -0,0 +1,21 @@
+    def group_params
+      params.require(:group).permit(:name, { :user_ids => [] })
+    end
+end
```
こちらもファイル末尾の改行にご注意。

sato11 commented on app/models/group.rb:
```diff
@@ -0,0 +1,6 @@
+class Group < ApplicationRecord
+
+  has_many :group_users, dependent: :destroy
```
グループ削除機能を実装してないのにこの記述があるのってどうだろう？

oYasuiHaruki replied:
削除機能がないのにつける意味はないですね
消しておきます！

sato11 commented on app/models/group.rb:
```diff
@@ -0,0 +1,6 @@
+class Group < ApplicationRecord
+
+  has_many :group_users, dependent: :destroy
+  has_many :users, :through => :group_users
```
ハッシュロケット使わないで！

sato11 commented on app/views/group/edit.html.erb:
```diff
@@ -0,0 +1,36 @@
+    </div>
+    <div class='chat-group-form__field clearfix'>
+      <div class='chat-group-form__field--left'>
+        <label class="chat-group-form__label" for="chat_group_チャットメンバーを追加">チャットメンバーを追加</label>
```
for 属性ってなにをしているんだっけ？

oYasuiHaruki replied:
for属性はlabelで関連させるid値を指定しているんですね
わからず使っていました
ご指摘ありがとうございます！

sato11 commented on app/views/group/edit.html.erb:
```diff
@@ -0,0 +1,36 @@
+<div class='chat-group-form'>
+  <h1>チャットグループ編集</h1>
+  <form class="edit_chat_group" id="edit_chat_group_22" action="/chat_groups/22" accept-charset="UTF-8" method="post"><input name="utf8" type="hidden" value="&#x2713;" />
```
form は全部 form_for を使って書きましょう！！
ヘルパー使わないのはたいへん非効率なので！

sato11 commented on app/views/top/index.html.haml:
```diff
@@ -1,42 +1,44 @@
-    %input{type: :"submit", value: :"Send", class: :"send"}
-    = fa_icon 'file-photo-o'
+      .group__header__icon
+        = link_to '/group/new' do
```
パス指定は prefix 使おう！！

sato11 reviewed (commented):
頑張って haml 導入しよう 💪

oYasuiHaruki committed:
idを1と指定したが、編集機能の追加ができた
次はid属性をtopページから選択できるようにする

oYasuiHaruki committed:
トップページのビューのscssの編集
・hoverをつけたほうがわかりやすいと思い、付け加えました

oYasuiHaruki committed:
gruopの編集機能の搭載
messageのコントローラーを追加

yasui83 commented:
# what
グループの編集機能を搭載した
・メッセージコントローラーの導入
・トップページにグループ名を表示させた
・トップページにグループにいるメンバーを表示させた

# why
編集機能を搭載するため

sato11 commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,28 @@
+    redirect_to("/")
+  end
+
+  def create
```
newの近くにcreateあった方が分かりやすそうだね

oYasuiHaruki replied:
その通りですね
なおしておきます！

sato11 commented on app/models/group.rb:
```diff
@@ -0,0 +1,6 @@
+
+  has_many :group_users
+  has_many :users, through: :group_users
+
```
バリデーションをつけましょう

sato11 commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,28 @@
+    redirect_to("/")
+  end
+
+private
```
ここインデント下げる必要なしです！

sato11 commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,28 @@
+
+  def create
+    Group.create(group_params)
+    redirect_to("/")
```
prefix で指定しよう！

sato11 commented on app/controllers/messages_controller.rb:
```diff
@@ -0,0 +1,9 @@
+  def index
+    @group = Group.all
+    @current_group = Group.find(params[:group_id])
+    @user = @current_group.group_users
```
変数名の単数複数に配慮してもっと上手く命名しましょう！

sato11 commented on app/controllers/messages_controller.rb:
```diff
@@ -0,0 +1,9 @@
+class MessagesController < ApplicationController
+
+  def index
+    @group = Group.all
```
ログイン中のユーザーが属しているグループだけを取れないかな？

sato11 commented on app/views/groups/edit.html.haml:
```diff
@@ -0,0 +1,22 @@
+.chat-group-form
+  %h1 チャットグループ編集
+  =form_for @group, method: :patch do |f|
```
httpメソッドの指定いるんだっけ？form_forの仕様を確認してみよう。

oYasuiHaruki replied:
引数で渡したレコードがすでに存在している場合は自動でpatchに変わる仕様でした！
消しておきます！

sato11 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,45 @@
+  .group
+    .group__header
+      .group__header__name
+        #{current_user.name}
```
式展開せずにhamlの書き方を採用しよう！

sato11 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,45 @@
+      .group__header__name
+        #{current_user.name}
+      .group__header__icon
+        = link_to "#{new_group_path}" , class: "icon1" do
```
ここも式展開必要かな？

sato11 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,45 @@
+        = link_to "#{edit_user_registration_path(current_user)}" , class: "icon2" do
+          = fa_icon 'cog'
+    .group__user
+      - @group.each do |group|
```
部分テンプレートを活用すれば each よりもスマートにかけるね。

sato11 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,45 @@
+      - @group.each do |group|
+        = link_to "#{group_messages_path(group)}", class: "group__user__list" do
+          .group__user__list--name
+            = simple_format(group.name)
```
このメソッドはなんのために使ってる？

oYasuiHaruki replied:
改行しないのに使う必要はなさそうですね
消しておきます！

sato11 reviewed (commented):
ちょっと変更ファイルが多くなっているから、次回からは糸つのブランチあたりの作業量にも配慮して実装できるようにしましょう！
例えば今回なら、グループ作成とグループ編集を分けて実装することで、よりスピーディーに開発&FBをできたのではないかなと思います！

oYasuiHaruki committed:
・redirect_toのパスをprefixで記入
・privateのインデントを揃える
・controllerで定めた変数の単数複数に注意して命名
・groupモデルにnameカラムにvalidationを設定
・form_forにmethod 書く必要がなかったので、消す
・部分テンプレートを作成
・式展開ではなく =　に変更

yasui83 commented:
# what
・redirect_toのパスをprefixで記入 
・privateのインデントを揃える
・controllerで定めた変数の単数複数に注意して命名
・groupモデルにnameカラムにvalidationを設定
・form_forにmethod 書く必要がなかったので、消す
・部分テンプレートを作成
・式展開ではなく =　に変更

# why
・コードを読みやすくするため
・validationを設定するため

oYasuiHaruki committed:
文字式展開を= に変換

sato11 commented on app/assets/javascripts/application.js:
```diff
@@ -13,4 +13,4 @@
 //= require jquery_ujs
 //= require turbolinks
-//= require_tree .
+// = require_tree .
```
なんのための変更？

sato11 commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,28 @@
+
+  def create
+    Group.create(group_params)
+    redirect_to("#{root_path}")
```
式展開必要？？

sato11 commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,28 @@
+
+  def update
+    group = Group.find(params[:id])
+    group.update(group_params)
```
バリデーションに引っかかった時ってどうなる？
更新できなかったときの例外処理があるとイイね！

sato11 replied:
こちらにも例外処理があるべきだね。
仮に現在の仕様で例外が絶対に発生し得ないとしても、将来的な拡張性を担保するためにあらかじめ例外時の動作を定義しておくのが良いエンジニアだね。

sato11 commented on app/controllers/messages_controller.rb:
```diff
@@ -0,0 +1,9 @@
+
+  def index
+    @group_users = GroupUser.where(user_id: current_user.id)
+    @current_group = Group.find(params[:group_id])
```
currentってつけなくてもよさそうだね

sato11 commented on app/models/application_record.rb:
```diff
@@ -1,3 +1,4 @@
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+
```
なんのための変更だろう？
本質的な変更以外はプルリクエストに含めなくてよさそうだね

sato11 commented on app/views/groups/new.html.haml:
```diff
@@ -0,0 +1,23 @@
+.chat-group-form
+  %h1 新規チャットグループ
+  =form_for @group do |f|
```
これ以下editと共通なら部分テンプレートにしちゃおう！

sato11 replied:
```
=form_for @group do |f|
  = render partial: "group"
```
↑ここも部分テンプレートに含められないかな？

sato11 commented on app/views/layouts/_group__user.html.haml:
```diff
@@ -0,0 +1,5 @@
+= link_to "#{group_messages_path(g.group)}", class: "group__user__list" do
```
ちょっとへるぱーの引数について整理してみよう。
式展開ってなんのためかな？

sato11 commented on app/controllers/messages_controller.rb:
```diff
@@ -0,0 +1,9 @@
+class MessagesController < ApplicationController
+
+  def index
+    @group_users = GroupUser.where(user_id: current_user.id)
```
最終的に g.group.users になるんだったら最初からグループモデルを呼び出せばいいんじょないかな？

oYasuiHaruki committed:
link_toの引数の書き方の変更

oYasuiHaruki committed:
flashメッセージを追加
コントローラーの編集
groupsのビューの共通部分を部分テンプレート化

yasui83 commented:
# what
flashメッセージの導入
コントローラーの記述の変更

# why 
グループを作成するときにnameを記入していない時のバリデーションを表示するため
コントローラーに記述する変数を無駄な処理ない変数に変えて、コードを見やすくするため

sato11 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,42 @@
+      .group__header__name
+        = current_user.name
+      .group__header__icon
+        = link_to (new_group_path) , class: "icon1" do
```
引数のカッコはあってもなくてもいいね。

sato11 replied:
icon1, icon2 っていうクラスって果たして管理しやすいでしょうか？
html では前後の文脈があるからわかるにしても、 css や js で扱いたい時に不便になりそうだから、具体的なクラス名を付与した方が良さそうだね！

sato11 commented on app/views/layouts/_group__user.html.haml:
```diff
@@ -0,0 +1,5 @@
+= link_to (group_messages_path(g)), class: "group__user__list" do
```
このファイルを layouts/ に配置したのはどんな意図がある？

oYasuiHaruki replied:
正直、どこに配置すればいいか分かりませんでした。
ただrenderメソッドでcolledctionを使う場合、この部分テンプレートをgroupsディレクトリ以下に保存することでエラーが起きずに出力することができました。
ただなぜgroupsディレクトリ以下に保存するとgroupと書いてeachと同じような働きをする理由が分かりません。
調べたのですが、あまりrenderのcollectionメソッドについての記述がなく分かりませんでした。
もしよかったら教えていただけると嬉しいです。

sato11 replied:
layouts/ は基本的に共通レイアウト（今回だと `application.html.haml` ）を配置する場所だと思ってくれれば 👌 
部分テンプレートの置き場としては、例えば `views/shared/_header.html` という形になったり、今回であれば `_form.html` はグループ用のフォームなので `views/groups/_form.html` だったりと柔軟に（ただしわかりやすい場所に）配置するのが正しいですね。

sato11 replied:
>renderメソッドでcolledctionを使う場合、この部分テンプレートをgroupsディレクトリ以下に保存することでエラーが起きずに出力することができました。
ただなぜgroupsディレクトリ以下に保存するとgroupと書いてeachと同じような働きをする理由が分かりません。

partial オプションで正しく指定しているにも関わらず、 `layouts/_group.html` では読み込まれなくて `groups/_group.html` だと読まれたってことでいいかな？
結論今回のケースでは @groups という変数が Groupモデルから導かれたリソースなので、
 `view/groups` 以下に探しに入ってくれるというRailsの仕様のはずなんですが、 `partial: 'layouts/group'` と明示的に書いたにも関わらず読み込まれなかったというのは謎ですね、、。
もう少し詳しく状況を教えてもらうことはできますか？

oYasuiHaruki replied:
色々触ってみたのですが、僕の勘違いでした！
layoutsファイル以下に保存しても問題なくいけました。
お騒がせしてすみません。

sato11 commented on app/views/groups/_group.html.haml:
```diff
@@ -0,0 +1,19 @@
+.chat-group-form__field.clearfix
```
`groups/_group.html.haml` ってファイル名は命名規則からいくと複数の group について繰り返し描画するような形になるね。
ちょうど `tweets/_tweet.html` と同じと思ってくれればわかりやすいと思う。ここで部分化したのは form 要素だから素直に _form.html とかでいいんじゃない？

oYasuiHaruki replied:
命名規則についてしっかり理解していなかったです！
今回は複数の要素を元にした部分テンプレートではないので、groupと書くべきじゃないですね。
formに書き換えておきます！

sato11 commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,34 @@
+
+  def create
+    @group = Group.create(group_params)
+    if @group.valid? == true
```
条件文に悩んでしまったかな？
「saveメソッドは返り値に真偽値をもつ」ということがヒントになります！もうひと押し頑張ろう！

ちなみに `@group.valid?` 自身が評価されるから `== true` はなくても大丈夫だね。
過不足のないコーディングをできるようになろう！

oYasuiHaruki replied:
一つ質問したいのですが、今回groupコントローラーのcreateアクションのところなんですが、
groupのmodelファイルにaccepts_nested_attributes_forをつけると
groupコントローラーのcreateアクションがエラーを起こします。
なぜ保存できないのか教えていただけないでしょうか。

ちなみになんですが、groupコントローラーのストロングパラメーターでuser_idsの記述をなくすとエラーは起きずにアクション自体は成功するんですが、データとして保存されないです。
なのでaccepts_attributes_forを書くと、ストロングパラメーターで指定した子モデルも一緒に保存しようとして、saveができないのかと思っています。
調べたのですが、いまいち分からなかったため質問させてもらっています。
お手数おかけしますが、よろしくお願いします。

sato11 replied:
>groupコントローラーのcreateアクションがエラーを起こします。
なぜ保存できないのか教えていただけないでしょうか。

エラーが起こるというのは常に一意的な理由があるという訳ではないはずだから、↑の情報から僕が何かを突き止めるということは非常に難しいです、、ごめんなさい 🙇 
「エラー」という言葉は漠然としているから、エラーメッセージやログをいただけるとそれに対してなんらかのコメントをすることは可能です！

sato11 replied:
ちなみに現時点では `accepts_nested_attributes_for` を使わなくても中間テーブルにレコードは保存されてるということでOKですか？
その部分が実装されていて動作が確認済みであれば実装としては問題ないはずです！

oYasuiHaruki replied:
はい！
現時点ではaccepts_nested_attributes_forを使わなくても中間テーブルにレコードを保存できている状況です！
なので実装としては問題ないと思います！

oYasuiHaruki replied:
エラーというよりは
https://gyazo.com/ecc450eba15480431cec0eebc51e73c8
ちゃんと名前も書いているのに、@group.saveが常にfalseになってしまうことの理由がわからなく質問させていただきました。
質問の仕方がよくなくてすみません。

sato11 replied:
OK、じゃあコントローラー内でpryで処理を止めて、以下の命令をコンソールでやってみましょう。
```pry
[1] > @group.save
# => error
[2] > @group.valid?
=> false
[3] > @group.errors
# => なんらかのエラーメッセージが可視化できるはず
```

oYasuiHaruki replied:
https://gyazo.com/7759d186faa20b85450b99bffe976992
こんな感じのエラー文でした！

sato11 commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,34 @@
+      redirect_to(root_path)
+    else
+      flash[:alert] = "グループ名を入力してください"
+      redirect_to(new_group_path)
```
一般的にアクションを差し戻す時はリダイレクトではなく render を使います！
これにはしっかりとした理由があるから、違いを調べて取り入れてみましょう！フラッシュメッセージの扱い方も変化します！

oYasuiHaruki replied:
renderは使うビューを指定するだけなので、生成したインスタンスはそのまま使うときにはrenderを使うという理解であっているでしょうか。
今回の場合は@groupにインスタンスを生成していて、そのインスタンスをビューで使用するので今回はrenderメソッドを使うという理解であっていますか。

oYasuiHaruki replied:
https://gyazo.com/67bd5a940895be14280243ceff09879f
フラッシュメッセージなのですが、このままでも表示できるのですが
扱い方が変わるというのはどういう意味でしょうか。

http://qiita.com/jnchito/items/94f3ad15128f88bf89d6
こんな記事を見かけたのですが、flashのままでflash.nowにしなくても表示できています。
https://gyazo.com/5956b58a332998687f30d9ff04f22cc3

sato11 replied:
>生成したインスタンスはそのまま使うときにはrenderを使う

多分理解としては悪くない線いってると思うんだけど、ちょっと定義が曖昧で心配だからしっかり説明できるようにしておきましょう！

- `redirect_to` の場合: HTTPリクエストが再度発行されるため、ルーティング/コントローラ/SQL etc. の処理が再施行される
- `render`の場合: #createアクションのリクエスト内でビューを描画するだけで完結する

これによってどんな利点があるかというと、
1. 処理に無駄がないぶん実行が速い
1. ユーザーの入力値が初期化されない（持ち越される）

の二つが大きいです！

sato11 replied:
>flashのままでflash.nowにしなくても表示できています。

結論表示はされます！ただ `now` を明示しないとUXの観点からあまり都合のよくない現象が起こるんだけど、なにか気づいたことはないかな、、？
自分でchatspaceを色々回遊してみるとすぐにわかると思うし、 now メソッドの定義もしっかり確認して見た方が良さそうかなと思います！

oYasuiHaruki replied:
http://l-light-note.hatenablog.com/entry/2014/09/25/205211
この記事に載っていました！
nowと記述することで１回だけflashメッセージを乗せることができるんですね

sato11 commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,42 @@
+        = link_to (edit_user_registration_path(current_user)) , class: "icon2" do
+          = fa_icon 'cog'
+    .group__user
+      - @groups.each do |g|
```
よほど自明な時以外は変数名として頭文字を使うのは避けた方がいいです、、。
ローカル変数として一時的に宣言するならともかく、今回は部分テンプレート側を読んだ時に g とだけ書いてあっても何が入っているのかわからず、結局呼び出し元の定義を確認しなければならなくなってしまっています。

またここで `.group__user` というクラス名を使っているけど、これだと user を描画しているように見えてしまうね。
`@groups` について扱うのであれば、クラス名も `.groups` とするのがもっとも素直でわかりやすいかと！

sato11 replied:
ちなみに each を使わずに部分テンプレートを繰り返し表示するオプションが render メソッドにはあったはずだけど、それを使うのがここでは一番スマートではないかな？

sato11 commented on app/views/top/index.html.haml:
```diff
@@ -1,42 +1,14 @@
-    %textarea{name: :"type", class: :"type", placeholder: :"type a message"}
-    %input{type: :"submit", value: :"Send", class: :"send"}
-    = fa_icon 'file-photo-o'
+.wrapper
```
ここも共通部分のようだから積極的に部分テンプレートとして管理すると非常に効率的だと思います！

sato11 commented on app/assets/stylesheets/_group.scss:
```diff
@@ -0,0 +1,96 @@
+      cursor: pointer;
+
+      &--add {
+        color: #38aef0;
```
この辺りのカラーコードって定義していなかったっけ？

sato11 reviewed (commented):
指摘箇所は多いですがだんだん良くなってきていると思います！
もうこのブランチもラストの追い込みのはずだから、あと一息頑張りましょう！

oYasuiHaruki committed:
・クラス名の変更
・見やすいようにインデントを揃える
・カラーコードの定義
・groupコントローラーで例外処理
・form_forを部分テンプレート化
・

yasui83 commented:
# what
・index.html.erbのクラス名の変更  
・見やすいようにインデントを揃える
・カラーコードの定義
・groupコントローラーで例外処理
・form_forを部分テンプレート化

# why
・分かりやすいクラス名にするため
・例外処理を行うため
・部分テンプレートして、コードを見やすくする

sato11 commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,40 @@
+  end
+
+  def update
+    @group = Group.find(params[:id])
```
ここと 20行目は同じ処理なので、 before_action を使って一つの処理にまとめてしまうことができるね！

sato11 commented on app/views/groups/_index.html.haml:
```diff
@@ -0,0 +1,10 @@
+.group__header
```
部分テンプレートの切り出し方として、 `.left-content` まで含めてしまっていいと思います！

それから前回と同様にファイルの命名についてだけど、 _index って名前ってあまり具体的ではないよね、、？
切り分けた部分が `.left-content` なのだから、 `_left.content.html` とか `sidebar.html` とかがいいのではないかなと思います！

oYasuiHaruki replied:
命名について甘いですね。
命名の時にもっと具体的かどうかを考えて、命名したいと思います！

sato11 commented on app/views/groups/_index.html.haml:
```diff
@@ -0,0 +1,10 @@
+    = link_to edit_user_registration_path(current_user) , class: "cog" do
+      = fa_icon 'cog'
+.groups
+  = render partial: "groups/group", collection: @groups
```
動作としてはこれでいいんだけど、実はまだ短く書くことができます！
まず collection オプションは黙ってても `groups/_group.html` を探しに入ってくれるので `partial` は不要です！
それから確かPictweetでやったんじゃないかと思うんだけど、単に `render @groups` と書いてみると何が起こるでしょう、、？

oYasuiHaruki committed:
・before_action,flash.nowを追加
・部分テンプレートを変更

yasui83 commented:
# what 
・before_action,flash.nowの追加
・部分テンプレートの編集

# why
・flash.nowにしないとflashメッセージが残ってしまうため
・コードの読みやすさを上げるため

sato11 commented on app/assets/stylesheets/_chat-space.scss:
```diff
@@ -159,6 +199,13 @@ body {
   }
 }
+
+
```
非常に細かいんだけど、ファイル末尾の改行は1行で必要十分なので、不要な文は残さないように心がけるとベターです！

sato11 commented on app/views/groups/_left-content.html.haml:
```diff
@@ -0,0 +1,11 @@
+    .group__header__icon
+      = link_to new_group_path , class: "pencil-square-o" do
+        = fa_icon 'pencil-square-o'
+      = link_to edit_user_registration_path(current_user) , class: "cog" do
```
font-awesome のクラス名にしないといけないかな？
リンクに貼るクラスとしては必ずしも適切でないような気がする 🤔

sato11 commented on app/assets/stylesheets/_chat-space.scss:
```diff
@@ -22,38 +22,76 @@ body {
+          display: inline-block;
+        }
+
+        .cog:hover{
```
`.pencil-square-o` と `.cog` のプロパティが重複しているの一元化できないかな？

oYasuiHaruki committed:
・iconにクラス名を変更

yasui83 commented:
# what
・クラス名をiconに変更

# why
・コードの可読性を上げるため

sato11 commented on app/assets/stylesheets/_chat-space.scss:
```diff
@@ -1,59 +1,89 @@
+          display: inline-block;
+        }
+
+        .icon:hover {
```
```
.icon {
  display: inline-block;
  &:hover {
    background: ...
  }
}
```
こうならないかな？

sato11 commented on app/assets/stylesheets/_chat-space.scss:
```diff
@@ -1,59 +1,89 @@
           font-size: 20px;
         }
+
+        .fa:hover {
```
↑が大丈夫ならこっちも！

sato11 commented on app/views/groups/_form.html.haml:
```diff
@@ -0,0 +1,20 @@
+=form_for @group do |f|
+  .chat-group-form__field.clearfix
+    .chat-group-form__field--left
+      %label.chat-group-form__label{for: :"chat_group_name"} グループ名
```
ごめん、ずっと見落としてしまっていたけどフォームのパーツについては form_for オブジェクト（ `f` ）のヘルパーメソッドを使って描いてください、、 🙇 
このファイルだけなのでそんなに難しくないと思う！！グッと綺麗になるのでお願いします！

oYasuiHaruki committed:
・コードの可読性を変更
・フォームのパーツについてform_forオブジェクトのヘルパーメソッドを使用

yasui83 commented:
# what
・hoverの書き方を変更
・form_forオブジェクトのヘルパーメソッドの使用

# why
・コードの可読性を上げるため
・form_forオブジェクトのヘルパーメソッドを使用するため

sato11 commented on app/assets/stylesheets/_flash.scss:
```diff
@@ -4,17 +4,15 @@
   text-align: center;
   color: white;
   margin: auto 0;
+  padding: 5px 0 5px;
```
ここ省略できない？

oYasuiHaruki replied:
paddingはいらないっていうことでしょうか？
確かに必要ないと思うので、記述を消しておきます！！

sato11 commented on app/assets/stylesheets/_group.scss:
```diff
@@ -0,0 +1,96 @@
+      float         : left;
+      width         : 30%;
+      padding-right : 20px;
+      padding-left  : 20px;
```
`padding: 0 20px;` の方がスマートかも

oYasuiHaruki replied:
直しておきました！

sato11 reviewed (commented):
OKです！画面のキャプチャだけ貼って共有してください！

yasui83 commented:
https://gyazo.com/d3fbfa30609c62f0ac4f956bb11d8de7
https://gyazo.com/46aa912c09982b13f706c3be77b0e20c
https://gyazo.com/d5a15de61dbec679b0b5f0a990f36c53
https://gyazo.com/f77122c04ce51e9fcfe0d47a3c3a2468
https://gyazo.com/2e8d37b63f452d8a7643107d73589d66
https://gyazo.com/a9593295507b087917f10c8ac6a9055a
https://gyazo.com/d3fbfa30609c62f0ac4f956bb11d8de7

oYasuiHaruki committed:
・paddingの書き換え

oYasuiHaruki committed:
・saveボタンの位置をUIの観点より配置

yasui83 commented:
https://gyazo.com/addc490f1da39ad1fa2ef1272c1667ad
# what
・saveボタンの位置の変更

# why
・UXの観点より
